engine/commands/cortex_upd_cmd.cc

```diff
@@ -99,73 +117,82 @@ void CortexUpdCmd::Exec(std::string v) {
                                                 extract_path.string());
 
-                  // remove the downloaded file
-                  // TODO(any) Could not delete file on Windows because it is currently hold by httplib(?)
-                  // Not sure about other platforms
-                  try {
-                    std::filesystem::remove(absolute_path);
-                  } catch (const std::exception& e) {
-                    CTL_WRN("Could not delete file: " << e.what());
-                  }
                   CTL_INF("Finished!");
                 });
             break;
           }
         }
       } catch (const nlohmann::json::parse_error& e) {
         std::cerr << "JSON parse error: " << e.what() << std::endl;
-        return;
+        return false;
       }
     } else {
       CTL_ERR("HTTP error: " << res->status);
-      return;
+      return false;
     }
   } else {
     auto err = res.error();
     CTL_ERR("HTTP error: " << httplib::to_string(err));
-    return;
+    return false;
   }
-#if defined(_WIN32)
-  auto executable_path = file_manager_utils::GetExecutableFolderContainerPath();
-  auto temp = executable_path / "cortex_tmp.exe";
-  remove(temp.string().c_str());  // ignore return code
-
-  auto src =
-      executable_path / "cortex" / kCortexBinary / (kCortexBinary + ".exe");
-  auto dst = executable_path / (kCortexBinary + ".exe");
-  // Rename
-  rename(dst.string().c_str(), temp.string().c_str());
-  // Update
-  CopyFile(const_cast<char*>(src.string().c_str()),
-           const_cast<char*>(dst.string().c_str()), false);
-  auto download_folder = executable_path / "cortex";
-  remove(download_folder);
-  remove(temp.string().c_str());
-#else
+
+  // Replace binay file
   auto executable_path = file_manager_utils::GetExecutableFolderContainerPath();
-  auto temp = executable_path / "cortex_tmp";
-  auto src = executable_path / "cortex" / kCortexBinary / kCortexBinary;
-  auto dst = executable_path / kCortexBinary;
-  if (std::rename(dst.string().c_str(), temp.string().c_str())) {
-    CTL_ERR("Failed to rename from " << dst.string() << " to "
-                                     << temp.string());
-    return;
-  }
-  try {
-    std::filesystem::copy_file(
-        src, dst, std::filesystem::copy_options::overwrite_existing);
-    std::filesystem::permissions(dst, std::filesystem::perms::owner_all |
-                                          std::filesystem::perms::group_all |
-                                          std::filesystem::perms::others_read |
-                                          std::filesystem::perms::others_exec);
-    std::filesystem::remove(temp);
-    auto download_folder = executable_path / "cortex/";
-    std::filesystem::remove_all(download_folder);
-  } catch (const std::exception& e) {
-    CTL_WRN("Something wrong happened: " << e.what());
-    return;
+  auto src = executable_path / "cortex" / "cortex-cpp" / GetCortexBinary();
```

Comment:
I think we standardized the paths to be `cortexcpp` (i.e. without hyphen) in #1115 - or perhaps I'm misunderstanding what this line does

Reply:
It's the cortex-binary name. Will update it